packages/atomic/src/components/facets-v1/facet-number-input/facet-number-input.tsx

```diff
@@ -31,44 +45,70 @@ export class FacetNumberInput {
       return;
     }
 
-    this.start = this.filterState.range?.start;
-    this.end = this.filterState.range?.end;
+    this.start = this.filterState.range?.start.toString() || '';
+    this.end = this.filterState.range?.end.toString() || '';
   }
 
   private apply() {
-    this.onApply && this.onApply();
+    this.applyInput.emit();
+    this.filter.setRange({
+      start: parseFloat(this.start),
+      end: parseFloat(this.end),
+    });
+  }
 
-    try {
-      this.filter.setRange({
-        start: this.start!,
-        end: this.end!,
-      });
-      this.isRangeInvalid = false;
-    } catch (error) {
-      this.bindings.engine.logger.error(error);
-      this.isRangeInvalid = true;
+  private validateInput(value: string) {
+    if (this.type === 'decimal') {
+      return value.replace(/(?!^-)[^0-9.]/g, '').replace(/(\..*)\./g, '$1');
     }
+
+    return value.replace(/(?!^-)[^0-9]/g, '');
```

Comment:
It's not clear to me what these regular expressions are trying to do.
It seems like on line 65, `10.1` would be converted to `101`?

If the goal is to remove the decimal, could `Math.floor` be a better option?